src/test/java/org/opensearch/flowframework/rest/FlowFrameworkRestApiIT.java

```diff
@@ -363,6 +363,281 @@ public void testCreateAndProvisionAgentFrameworkWorkflow() throws Exception {
         assertBusy(() -> { getAndAssertWorkflowStatusNotFound(client(), workflowId); }, 30, TimeUnit.SECONDS);
     }
 
+    public void testReprovisionWorkflow() throws Exception {
+        // Begin with a template to register a local pretrained model
+        Template template = TestHelpers.createTemplateFromFile("registerremotemodel.json");
+
+        // Hit Create Workflow API to create agent-framework template, with template validation check and provision parameter
+        Response response;
+        if (!indexExistsWithAdminClient(".plugins-ml-config")) {
+            assertBusy(() -> assertTrue(indexExistsWithAdminClient(".plugins-ml-config")), 40, TimeUnit.SECONDS);
+            response = createWorkflowWithProvision(client(), template);
+        } else {
+            response = createWorkflowWithProvision(client(), template);
```

Comment:
This can be simplified by moving this creation outside the `if` conditional and only using it to wait on the ml config index.   

Which seems somewhat of a common step for almost all our tests, can we add it in the `waitToStart()` section with the `@Before` annotation, perhaps replacing my hacked "wait 25 seconds" workaround there?

Reply:
Sure Ill make this modification